mdc_rpt_physical_worksheet/report/report_xlsx.py

```diff
@@ -0,0 +1,205 @@
+        f_dataF = workbook.add_format(f_cells["dataF"])
+        f_dataY = workbook.add_format(f_cells["dataY"])
+        f_dataL = workbook.add_format(f_cells["dataL"])
+        f_data2d  = workbook.add_format(f_cells["data2d"])
+        f_data2dY= workbook.add_format(f_cells["data2dY"])
+
+        # Set columns widths
+        sheet.set_column('A:AJ', 13)
+        sheet.set_column('B:B', 40) # Employee name
+
+        # write logo
+        logo_file_name = False
+        binary_logo = self.env['res.company'].sudo().search([]).logo_web
+        if binary_logo:
+            fp = tempfile.NamedTemporaryFile(delete=False)
+            fp.write(bytes(base64.b64decode(binary_logo)))
+            fp.close()
+            logo_file_name = fp.name
+        else:
+            # TODO enhance default logo path recovery
+            logo_file_name = "../addons/manufacture/mdc/report/logo.png"
+
+        sheet.insert_image('A1', logo_file_name, {
+            'x_offset': 18, 'y_offset': 18, 'x_scale': 0.9, 'y_scale': 0.5})
+
+        # write Title
+        sheet.write(1, 2, _("PHYSICAL WORKSHEET REPORT"), f_title)
+
+        # write Filter
+        sheet.write(2, 2, _("Date From:"), f_filter)
+
+        header_row=5
+        header_row_str = str(header_row+1)
+        # write column header ----------------------------------------
+        sheet.write('A' + header_row_str, _("Employee Code"), f_header)
+        sheet.write('B' + header_row_str, _("Employee Name"), f_header)
+        sheet.write('C' + header_row_str, _("Workstation"), f_header)
+        sheet.write('D' + header_row_str, _("Date"), f_header)
+        sheet.write('E' + header_row_str, _("Total Hours"), f_header)
+        sheet.write('F' + header_row_str, _("Time In 1"), f_header)
+        sheet.write('G' + header_row_str, _("Time Out 1"), f_header_ind)
+        sheet.write('H' + header_row_str, _("Time In 2"), f_header)
+        sheet.write('I' + header_row_str, _("Time Out 2"), f_header_ind)
+        sheet.write('J' + header_row_str, _("Time In 3"), f_header)
+        sheet.write('K' + header_row_str, _("Time Out 3"), f_header_ind)
+        sheet.write('L' + header_row_str, _("Time In 4"), f_header)
+        sheet.write('M' + header_row_str, _("Time Out 4"), f_header_ind)
+        sheet.write('N' + header_row_str, _("Time In 5"), f_header)
+        sheet.write('O' + header_row_str, _("Time Out 5"), f_header_ind)
+
+        # -------------------------------------------------------
+        # write data rows
+        row=header_row
+        wemployee_code = ''
+        wworksheet_date = ''
+        wtotal_hours = 0
+        wstart_datetime = None
+        wend_datetime = None
+        wnumcol = 0
+        wrowWithErrors=False
+        # Filters ---------
+        wstart_date = False
+        wend_date = False
+        wshift_filter = ''
+        wshift_filter_uniq = True
+        # -----------------
+
+        for obj in rpt_physical_worksheet:
```

Comment:
In order to fix (1), `obj.worksheet_date` should be recalculated starting each `for` iteration:

```suggestion
        for obj in rpt_physical_worksheet:
            obj_worksheet_date = self.convert_UTC_to_TZ(obj.worksheet_date)
```
Then, replace any `obj.worksheet_date` reference to `obj_worksheet_date`.
If this changed is made, no more UTC to TZ conversions are needed later, e.g. the time conversion below (line 182).
With this change date management becomes fully coherent.
What do you think?

Reply:
I have a problem with this field. This is fields.Date not fields.Datetime

Reply:
I´m use only obj_worksheet_datetime in xlsx, but I'm not sure it's the right thing.
What do we do with worksheet that start today and end tomorrow? 
Right now it doesn't look very good. 
Try report for 02/03/2020 and sift 2. Look at Anwar, Shawaiz...